lib/gui_rpc_client.cpp

```diff
@@ -455,16 +465,30 @@ int read_gui_rpc_password(char* buf, string& msg) {
                 return ERR_FOPEN;
             }
         } else {
-            char buf2[MAXPATHLEN];
-            if (!getcwd(buf2, MAXPATHLEN)) {
-                strcpy(buf2, "");
+            // no config file; look in default data dir
+            //
+            sprintf(path, "%s/%s", LINUX_DEFAULT_DATA_DIR, GUI_RPC_PASSWD_FILE);
+            f = fopen(path, "r");
+            if (!f) {
+                if (errno == EACCES) {
+                    sprintf(msg_buf,
+                        "%s exists but can't be read.  Check the file permissions.",
```

Comment:
Please add another "in %s" and ", LINUX_DEFAULT_DATA_DIR". Otherwise very nice, many thanks!